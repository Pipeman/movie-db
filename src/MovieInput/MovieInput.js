import React from "react";

const MovieInput = ({ value, handleChange }) => (
  <div className="movie-input-copntainer">
    <label htmlFor="movie-name">Movie to search</label>
    <input
      id="movie-name"
      className="movie-input"
      type="text"
      placeholder={value}
      onChange={handleChange}
    />
  </div>
);

export default MovieInput;