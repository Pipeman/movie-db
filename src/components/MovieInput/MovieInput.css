.movie-input {
  -webkit-appearance: none;
  border: none;
  border-bottom: .1rem blue solid;
  height: 2em;
  width: 100%;
}