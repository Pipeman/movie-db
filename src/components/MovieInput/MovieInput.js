import React from "react";

const MovieInput = ({ value, handleChange }) => (
  <div className="movie-input-copntainer">
    <input
      id="movie-name"
      className="movie-input"
      type="text"
      placeholder="Movie to search"
      onChange={handleChange}
    />
  </div>
);

export default MovieInput;