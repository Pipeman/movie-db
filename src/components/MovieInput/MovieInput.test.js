import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import MovieInput from "./MovieInput";

it("shoudld render the component correctly", () => {
  const testHandler = () => {};
  const wrapper = shallow(
    <MovieInput
      handleChange={testHandler}
    ></MovieInput>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("shoudld call the handler on change", () => {
  const testHandler = jest.fn();
  const wrapper = shallow(
    <MovieInput
      handleChange={testHandler}
    ></MovieInput>
  );
  wrapper.find("input").simulate("change");
  expect(testHandler).toHaveBeenCalledTimes(1);
});