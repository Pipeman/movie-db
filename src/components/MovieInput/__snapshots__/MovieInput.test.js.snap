// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoudld render the component correctly 1`] = `
<div
  className="movie-input-copntainer"
>
  <input
    className="movie-input"
    id="movie-name"
    onChange={[Function]}
    placeholder="Movie to search"
    type="text"
  />
</div>
`;
